main.go

```diff
@@ -471,7 +452,7 @@ func (c *ipv6Configuration) handleDNSRequest(p *proxy.Proxy, ctx *proxy.DNSConte
 // NewTLSConfig returns a TLS config that includes a certificate
 // Use for server TLS config or when using a client certificate
 // If caPath is empty, system CAs will be used
-func newTLSConfig(options *Options) (*tls.Config, error) {
+func newTLSConfig(options *Options, auth bool) (*tls.Config, error) {
```

Comment:
I don't think "auth" is necessary, you can simply check if `TLSAuthCertPath` is specified or not